src/Riok.Mapperly/Descriptors/MappingBuilders/EnumMappingBuilder.cs

```diff
@@ -83,6 +107,20 @@ private static TypeMapping BuildNameMapping(MappingBuilderContext ctx, bool igno
             ctx.ReportDiagnostic(DiagnosticDescriptors.EnumNameMappingNoOverlappingValuesFound, ctx.Source, ctx.Target);
         }
 
+        var missingSourceMembers = sourceFieldsByName.Select(item => item.Key).Except(enumMemberMappings.Select(mapping => mapping.Key));
+
+        if (missingSourceMembers.Any())
+        {
+            ctx.ReportDiagnostic(DiagnosticDescriptors.NotAllSourceEnumValuesMapped, missingSourceMembers.First(), ctx.Source, ctx.Target);
+        }
+
+        var missingTargetMembers = targetFieldsByName.Select(item => item.Key).Except(enumMemberMappings.Select(mapping => mapping.Value));
+
+        if (missingTargetMembers.Any())
+        {
+            ctx.ReportDiagnostic(DiagnosticDescriptors.NotAllTargetEnumValuesMapped, missingTargetMembers.First(), ctx.Target, ctx.Source);
+        }
```

Comment:
A few comments on this:

- Your proposed solution replicates the mapping done earlier with it's own implementation. I think this diagnostic should be based on the actual mapping created. 
- My proposed solution does in fact work for ignoreCase=true since it works with the actual mappings created. 
- I like that your solution reports ALL missing members instead of only the first

Reply:
You are totally correct, my bad! I'm also fine with your solution. Probably just change the any to a foreach.